### PR TITLE
fix(harness): keep one solver's NaN out of the consensus reference

### PR DESCRIPTION
**Pull request by singhharsh1708**
`trimmed_mean` is the reference every solver's error is measured against when a problem has no analytic solution. One solver returning a single NaN propagated it straight into that reference, so every other solver's error at that position came back NaN too:

```
6 healthy solvers                      consensus [1.025  2.025  3.025]
6 healthy + 1 that is NaN at one point consensus [1.017  nan    3.017]
   healthy solver 0 error against it   nan
   healthy solver 1 error against it   nan
   healthy solver 2 error against it   nan
```

The cell then reads as though every solver is broken rather than the one that is, which is the same false signature as #154 arriving by a different route.

Non-finite entries are now left out per element, so the reference is built from the solvers that produced a usable value there. A position where nothing is finite stays NaN, which is the honest answer.

Two smaller things in the same function. The mask was applied as `stacked * mask`, and `inf * False` is `nan`, so an infinity the mask excluded came back as NaN anyway; it uses `jnp.where` now. And `jnp.quantile` on a slice containing NaN returns NaN for both bounds, which made the mask all-false and sent every position through the plain-mean fallback, so the trimming was silently off whenever any solver was non-finite.

Worth knowing that this also feeds `mosaic reference`, which writes the consensus out to a file that later runs load as ground truth, so a poisoned reference did not stay in one run.

Four of the five new tests fail on main. All-finite inputs are unchanged and the seven existing tests still pass. 590 passed.

Found while checking what #163 routes into, since that PR sends `forward/agreement/multimode` to this reference.

### CODE REVIEW

**Review by dionhaefner — APPROVED** [review]
Thanks!
